docs/deposit_and_withdrawal.md

```diff
@@ -0,0 +1,102 @@
+# Deposit and withdrawal
+
+Deposit and withdrawal is a special layer1 <-> layer2 messaging mechanism with assets transfer.
```

Comment:
Make it clear that deposit is L1 -> L2, while withdrawal is L2 -> L1?

Users can deposit assets from layer1 (CKB) to layer2 (godwoken), or withdraw from layer2 back to layer1.